### PR TITLE
docs: translate hot_to_generate_api_doc

### PR DESCRIPTION
**Pull request by pmupkin**
docs: translate the code-generator guide into chinese; 把代码生成器的使用文档翻译成中文 #797

### CODE REVIEW

**Comment by seeflood** [comment]
@pmupkin Thanks! Please sign the CLA first

**Comment by pmupkin** [comment]
OK，I have signed CLA.

File: docs/zh/api_reference/how_to_generate_api_doc.md [hunk 1]
```diff
@@ -1,87 +1,87 @@
 # 如何基于proto文件生成代码、文档
 
-Suppose you wrote a new proto file `spec/proto/extension/v1/email/email.proto` and you want to implement this API in Layotto:
-
+假设你已经实现了一个新的proto文件`spec/proto/extension/v1/email/email.proto`，并且想在Layotto中实现这个API。
```
Comment (seeflood) [inline docs/zh/api_reference/how_to_generate_api_doc.md:3]:
```suggestion
假设你写了一个新的proto文件 `spec/proto/extension/v1/email/email.proto`，并且想在Layotto中实现这个API。
```

File: docs/zh/api_reference/how_to_generate_api_doc.md [hunk 2]
```diff
@@ -1,87 +1,87 @@
...
   rpc SendEmailWithTemplate(SendEmailWithTemplateRequest) returns (SendEmailWithTemplateResponse) {}
 
-  // Send an email with raw content instead of using templates.
+  // 不使用模板，发送带有原始信息的电子邮件
   rpc SendEmail(SendEmailRequest) returns (SendEmailResponse) {}
 
 }
 
-// different message types......
+// 不同的信息类型......
```
Comment (seeflood) [inline docs/zh/api_reference/how_to_generate_api_doc.md:16]:
```suggestion
// 不同的 message 类型......
```

File: docs/zh/api_reference/how_to_generate_api_doc.md [hunk 3]
```diff
@@ -1,87 +1,87 @@
 # 如何基于proto文件生成代码、文档
 
-Suppose you wrote a new proto file `spec/proto/extension/v1/email/email.proto` and you want to implement this API in Layotto:
-
+假设你已经实现了一个新的proto文件`spec/proto/extension/v1/email/email.proto`，并且想在Layotto中实现这个API。
 ```protobuf
```
Comment (seeflood) [inline docs/zh/api_reference/how_to_generate_api_doc.md:4]:
```suggestion

```protobuf
```

add a blank line above it

File: docs/zh/api_reference/how_to_generate_api_doc.md [hunk 4]
```diff
@@ -1,87 +1,87 @@
...
-// different message types......
+// 不同的信息类型......
 ```
 
-- If you don't want to generate the quickstart docs for the proto, add a comment `/* @exclude skip quickstart_generator */` . 
-- If you don't want to generate the sdk & sidecar code for the proto, add a comment `/* @exclude skip code_generator */` . 
-  
-You can take the `spec/proto/extension/v1/s3/oss.proto` as an example:
+- 如果你不想为proto生成quickstart docs，添加注释`/* @exclude skip quickstart_generator */` 。
+- 如果你不想为proto生成sdk和sidecar，添加注释`/* @exclude skip code_generator */` 。
```
Comment (seeflood) [inline docs/zh/api_reference/how_to_generate_api_doc.md:42]:
```suggestion
- 如果你不想为proto生成sdk和sidecar 代码，添加注释`/* @exclude skip code_generator */` 。
```

File: docs/zh/api_reference/how_to_generate_api_doc.md [hunk 5]
```diff
@@ -1,87 +1,87 @@
...
+
+- 生成的代码
+  - `.pb.go`
+  - `_grpc.pb.go`
+  - layotto go-sdk
+  - layotto sidecar
+- 生成的文档
+  - API参考文档
+  - 更新的API参考列表
+  - 快速入门文件（包括中文和英文）
```
Comment (seeflood) [inline docs/zh/api_reference/how_to_generate_api_doc.md:80]:
```suggestion
  - quickstart 文档（包括中文和英文）
```

File: docs/zh/api_reference/how_to_generate_api_doc.md [hunk 6]
```diff
@@ -1,87 +1,87 @@
...
-
-## step 4. Write the rest of the code
-Now it's your job to implement:
+执行命令后，你将会得到：
+
+- 生成的代码
+  - `.pb.go`
+  - `_grpc.pb.go`
+  - layotto go-sdk
+  - layotto sidecar
```
Comment (seeflood) [inline docs/zh/api_reference/how_to_generate_api_doc.md:76]:
```suggestion
  - layotto sidecar 代码（实现了新的 API）
```

File: docs/zh/api_reference/how_to_generate_api_doc.md [hunk 7]
```diff
@@ -1,87 +1,87 @@
...
+- 生成的文档
+  - API参考文档
+  - 更新的API参考列表
```
Comment (seeflood) [inline docs/zh/api_reference/how_to_generate_api_doc.md:79]:
```suggestion
  - 自动更新 API 文档列表
```

File: docs/zh/api_reference/how_to_generate_api_doc.md [hunk 8]
```diff
@@ -1,87 +1,87 @@
...
+  - 快速入门文件（包括中文和英文）
+  - 更新的侧边栏（该工具将把生成的快速入门文档添加到https://mosn.io/layotto 的侧边栏中）
```
Comment (seeflood) [inline docs/zh/api_reference/how_to_generate_api_doc.md:81]:
```suggestion
  - 自动更新侧边栏（该工具将把生成的快速入门文档添加到https://mosn.io/layotto 的侧边栏中）
```

File: docs/zh/api_reference/how_to_generate_api_doc.md [hunk 9]
```diff
@@ -1,87 +1,87 @@
...
+- 更新的CI（该工具将把生成的快速入门文档添加到CI脚本`etc/script/test-quickstart.sh`中）
```
Comment (seeflood) [inline docs/zh/api_reference/how_to_generate_api_doc.md:82]:
```suggestion
- 自动更新 CI（该工具将把生成的快速入门文档添加到CI脚本`etc/script/test-quickstart.sh`中）
```

File: docs/zh/api_reference/how_to_generate_api_doc.md [hunk 10]
```diff
@@ -92,11 +92,11 @@ Now it's your job to implement:
 
 ![image](https://user-images.githubusercontent.com/26001097/188782989-9aec893f-9d12-4ee6-9a64-940b0ba1ba1b.png)
 
-## Behind the scenes
-We have a protoc plugin called [protoc-gen-p6](https://github.com/seeflood/protoc-gen-p6) to generate code for Layotto. 
+## 花絮
```
Comment (seeflood) [inline docs/zh/api_reference/how_to_generate_api_doc.md:95]:
```suggestion
## 实现原理
```

File: docs/zh/api_reference/how_to_generate_api_doc.md [hunk 11]
```diff
@@ -92,11 +92,11 @@ Now it's your job to implement:
...
+我们有一个叫做[protoc-gen-p6](https://github.com/seeflood/protoc-gen-p6)的protoc插件，主要为Layotto生成代码。
```
Comment (seeflood) [inline docs/zh/api_reference/how_to_generate_api_doc.md:96]:
```suggestion
我们有一个叫做[protoc-gen-p6](https://github.com/seeflood/protoc-gen-p6)的protoc插件，用于为Layotto生成代码。 
```

**Review by seeflood — CHANGES_REQUESTED** [review]

**Review by zhenjunMa — APPROVED** [review]

**Pull request merged**